### ISSUE TITLE
The Database of Odorant Responses (DoOR.data)

### ISSUE THREAD

**Issue by Dahaniel** [ropensci/onboarding#35]
- 1. What does this package do? (explain in 50 words or less)
  
  The data package of the DoOR project. The data package provides all published odor responses recorded from olfactory sensory neurons of the fruit fly _Drosophila_.
- 1. Paste the full DESCRIPTION file inside a code block below.

```
Package: DoOR.data
Type: Package
Title: Integrating Heterogeneous Odorant Response Data into a Common Response
    Model: A DoOR to the Complete Olfactome
Version: 2.0.0
Date: 2016-02-25
Authors@R: c(
    person("Daniel", "Münch", email = "daniel.muench@uni-konstanz.de", role = c("aut", "cre")),
    person("C. Giovanni", "Galizia", email = "giovanni.galizia@uni-konstanz.de", role = "aut"),
    person("Shouwen", "Ma", role = "aut"),
    person("Martin", "Strauch", role = "aut"),
    person("Anja", "Nissler", role = "aut"))
URL: http://neuro.uni-konstanz.de/DoOR
BugReports: https://github.com/Dahaniel/DoOR.data/issues
Description: This is a data package providing Drosophila odorant response data for
    DoOR.functions. See http://dx.doi.org/10.1093/chemse/bjq042 for the original
    publication and http://dx.doi.org/10.1038/srep21841 regarding version 2.0.
License: CC BY-SA 4.0
Encoding: UTF-8
Imports:
    utils
Depends:
    R (>= 3.2.0)
Suggests:
    DoOR.functions (>= 2.0.0)
Remotes:
    Dahaniel/DoOR.functions@v2.0.0
RoxygenNote: 5.0.1

```
- 3. URL for the package (the development repository, not a stylized html page)

https://github.com/Dahaniel/DoOR.data
- 4. What data source(s) does it work with (if applicable)?

The package itself provides the data either extracted from publications or provided by the publication authors.
- 5. Who is the target audience?

Scientists working on olfactory coding, modelers and physiologists.
- 6. Are there other R packages that accomplish the same thing? If so, what is different about yours?

No, to our knowledge DoOR is the only project providing this kind of data and using this approach to integrate heterogeneous data.
- 7. Check the box next to each policy below, confirming that you agree. These are mandatory.
  - [x] This package does not violate the Terms of Service of any service it interacts with.
  - [x] The repository has continuous integration with Travis CI and/or another service
  - [ ] The package contains a vignette
  - [x] The package contains a reasonably complete README with `devtools` install instructions
  - [ ] The package contains unit tests
  - [x] The package only exports functions to the NAMESPACE that are intended for end users
- 8. Do you agree to follow the [rOpenSci packaging guidelines](https://github.com/ropensci/packaging_guide)? These aren't mandatory, but we strongly suggest you follow them. If you disagree with anything, please explain.
  - [ ] Are there any package dependencies not on CRAN? - _no_
  - [x] Do you intend for this package to go on CRAN?
  - [x] Does the package have a CRAN accepted license?
  - [ ] Did `devtools::check()` produce any errors or warnings? If so paste them below. - _no_
- 1. Please add explanations below for any exceptions to the above:
  - The package does not contain a vignette as it only provides data. All the functionality is comprehensively explained in the vignettes in the corresponding data package (see other onboarding).
  - I think unit tests do not make much sense here as the package solely provides data.
  - Travis has problems as the DoOR.data package suggests the DoOR.functions package. Adding DoOR.functions to "Remotes:" did not help as it depends on DoOR.data...
- 1. If this is a resubmission following rejection, please explain the change in circumstances.

**Comment by sckott** [comment]
Thanks for your submission, seeking reviewers

**Comment by noamross** [comment]
Reviewers: @RemkoDuursma 
Due date: 2016-05-09

**Comment by sckott** [comment]
@RemkoDuursma 
Due date: 2016-05-09 - hey there, it's been 16 days, please get your review in by May 08, thanks :smiley_cat: (ropensci-bot)

**Comment by RemkoDuursma** [comment]
This package includes data on olfactory responses of Drosophila. I do now know anything about this kind of data. I have reviewed this package only in terms of the R code, functionality, interface, and documentation.

The package only includes data as a set of dataframes, and no other functions. All data is loaded into the current workspace with `loadData()`. I find this choice of data porting somewhat unfortunate, as it simply loads all included datasets into the current workspace, potentially causing conflicts and certainly causing a messy workspace. But perhaps there is a good reason to do this, as I am not sure how these kind of data are used in analyses.

The 'standard' way of loading data is also available, though it seems not to be documented. For example we can do, `data("Or94b")` to load that particular dataset into the current workspace. The help page for `loadData` should show examples of this (and it should be renames `load_all_DoOr_data`, or similarly descriptive, see below).

The choice of data storage, in binary format (`Rdata` in data/) is fine in terms of storage space but prevents meaningful version control on the data. It is not clear to me (again due to my ignorance) if the underlying datasets are prone to change over time, or whether this is an unchanging database. If there is change to be expected within datasets, a different format would be preferable where incremental changes can be logged with version control.

I found the documentation to be extremely concise. For example the help file for `?data.format` only states "DoOr response data template", and all other help files are similarly brief. Surely more information is needed in order to use the package effectively (it would not be sufficient in my mind if this information can be found elsewhere).

The function names should follow protocol. Snake case is preferred, and the package should avoid datasets with very generic names (`response.matrix`,`dataset.info`,`data.format`, etc.). I would certainly avoid the use of `.` in names of functions to avoid possible S3 class dispatches.

In summary the package contains data only (I cannot judge whether these data are useful or broadly applicable), but some should be renamed, and documentation improved.

**Comment by sckott** [comment]
@RemkoDuursma 
Due date: 2016-05-09 - hey there, it's been 21 days, please get your review in soon, thanks :smiley_cat: (ropensci-bot)

**Comment by sckott** [comment]
hey there, for full openness - the bot commented above that a review wasn't in - but of course it is in. The offending line is here https://github.com/ropenscilabs/heythere/blob/master/helpers.rb#L41 - so we obviously need a more sophisticated way to tell if a review is submitted than how long a comment is.

**Comment by RemkoDuursma** [comment]
I can also conclude that my review is the shortest yet! It should have been
at least 3000 words :)

-------------------------------------------------
Remko Duursma
Senior Lecturer

Hawkesbury Institute for the Environment
Western Sydney University
Hawkesbury Campus, Richmond

Office: +61(0)2 4570 1806
Mobile: +61 (0)4 22 096908
www.remkoduursma.com

**Comment by sckott** [comment]
@RemkoDuursma well it was a data only a package, so short is okay in this case.

**Comment by sckott** [comment]
@Dahaniel 

* It would be nice to see better documentation of datasets here https://github.com/Dahaniel/DoOR.data/blob/master/R/DoOR.data-package.R - that is, describe what each column name is and what data the column has, and any units if applicable
* Also if datasets were from different sources 
* Here https://github.com/Dahaniel/DoOR.data/blob/master/R/loadData.R#L7 - I think there is a typical notation for email, like `\email{foo@bar.com}`

**Comment by Dahaniel** [comment]
First of all thank you very much for taking the time to review our package. 

I agree that the datasets need more documentation and I will look into that. Regarding the other points:

* **loading to current workspace**: I know this might be a problem and I am looking into possibilities to change that. We are doing that because the user should be able to easily and directly modify the loaded datasets (e.g. add or exclude rows or columns), and then rerun analysis tools (provided by `DoOR.functions`). Separate environments were not easily visible to the user. Any advice would be appreciated.

* **binary format** We initially stored the data as csv for exactly the reason of being able to track changes. We switched to binary to save some space. I would happily switch back to .csv. With .csv, could there be problems arising across operating systems e.g. regarding encoding of line-breaks?

* **function names** We tried to follow the [Google R style-guide](https://google.github.io/styleguide/Rguide.xml) i.e. camel code for function names and variable names separated by dots. In general I refrained from changing names as people are used to the established names. But I guess we could adapt all names to ROpenSci guide lines and make an announcement on the webpage and maybe print an message upon loading the package.


@sckott regarding email annotations: are you referring to the double @ sign? I think I copied that from somewhere but I just tested it and roxygen seems to convert both to proper email addresses in the help file.

**Comment by sckott** [comment]
@Dahaniel Ah, okay, if it works then all good

**Comment by noamross** [comment]
1) While snake case functions are preferred, for a package with existing users and consistent style other styles are fine.  If you wish to change function names to make them less generic and avoid `.`, I suggest using both in the next version of the package, with documentation/messages that the old versions are deprecated, and then removing the deprecated names in a future update.

2) In general, CSV should be interoperable between systems. Interoperability issues most commonly arise when Excel is somehow involved, as it encodes line breaks inconsistently across systems/versions.

As to loading in the workspace, I'd await @RemkoDuursma's review of DoOR.functions for his thoughts on the relationship between the two.

**Comment by sckott** [comment]
@RemkoDuursma 
Due date: 2016-05-09 - hey there, it's been 36 days, please get your review in soon, thanks :smiley_cat: (ropensci-bot)

**Comment by sckott** [comment]
@RemkoDuursma 
Due date: 2016-05-09 - hey there, it's been 37 days, please get your review in soon, thanks :smiley_cat: (ropensci-bot)

**Comment by sckott** [comment]
@RemkoDuursma 
Due date: 2016-05-09 - hey there, it's been 44 days, please get your review in soon, thanks :smiley_cat: (ropensci-bot)

**Comment by sckott** [comment]
@Dahaniel where are we at here?  Anything we can help with?

**Comment by Dahaniel** [comment]
@sckott sorry, I did not find the time to apply the changes yet.
I will change data back to csv, work on better documentation and explore ways to load data not to current workspace. 
Regarding function names I like the idea of making a slow transition, having old and new names work for a while and prompting a message explaining that. Is there an easy way to implement different names for the same function? Like defining an alias or similar?

**Comment by sckott** [comment]
You can simply do e.g., https://github.com/ropensci/RNeXML/blob/master/R/nexml_read.R#L70 and then use `@aliases` tag https://github.com/ropensci/RNeXML/blob/master/R/nexml_read.R#L8 to make sure it's linked and exported

**Comment by Dahaniel** [comment]
Please see: https://github.com/ropensci/onboarding/issues/34#issuecomment-284765039

**Comment by sckott** [comment]
@Dahaniel 

- R check reveals many problems: you need to Rbuildignore `.travis.yml`/`CHANGELOG.md` (but you should also have a `NEWS.md` or `NEWS` as that's more typical of R packages), `Undefined global functions or variables: data` - and more 
- there' a note about compressing your data better
- you can still write minimal tests even though this is a data pkg, e.g. load data, and make sure data is of right dimensions, etc. 

Some of R check output

```
   Found the following hidden files and directories:
     .travis.yml
   These were most likely included in error. See section ‘Package
   structure’ in the ‘Writing R Extensions’ manual.

   CRAN-pack does not know about
     .travis.yml
✔  checking for portable file names
✔  checking for sufficient/correct file permissions
✔  checking whether package ‘DoOR.data’ can be installed
✔  checking installed package size
✔  checking package directory
✔  checking DESCRIPTION meta-information
N  checking top-level files
   Non-standard file/directory found at top level:
     ‘CHANGELOG.md’
✔  checking for left-over files
✔  checking index information
✔  checking package subdirectories
✔  checking R files for non-ASCII characters
✔  checking R files for syntax errors
✔  checking whether the package can be loaded
✔  checking whether the package can be loaded with stated dependencies
✔  checking whether the package can be unloaded cleanly
✔  checking whether the namespace can be loaded with stated dependencies
✔  checking whether the namespace can be unloaded cleanly
✔  checking use of S3 registration
✔  checking dependencies in R code
✔  checking S3 generic/method consistency
✔  checking replacement functions
✔  checking foreign function calls
N  checking R code for possible problems
   loadData: no visible global function definition for ‘data’
   load_door_data: no visible global function definition for ‘data’
   Undefined global functions or variables:
     data
   Consider adding
     importFrom("utils", "data")
   to your NAMESPACE file.

   Found the following calls to data() loading into the global environment:
   File ‘DoOR.data/R/load_door_data.R’:
     data(list = all.data$results[, 3])
   See section ‘Good practice’ in ‘?data’.
✔  checking Rd files
✔  checking Rd metadata
✔  checking Rd line widths
✔  checking Rd cross-references
✔  checking for missing documentation entries
✔  checking for code/documentation mismatches
✔  checking Rd \usage sections
✔  checking Rd contents
✔  checking for unstated dependencies in examples
✔  checking contents of ‘data’ directory
✔  checking data for non-ASCII characters
W  checking data for ASCII and uncompressed saves

     Note: significantly better compression could be obtained
           by using R CMD build --resave-data
                       old_size new_size compress
     door_AL_map.RData    514Kb    396Kb       xz
```

**Comment by Dahaniel** [comment]
Thanks @sckott.

> * R check reveals many problems: you need to Rbuildignore
.travis.yml/CHANGELOG.md (but you should also have a NEWS.md or NEWS as that's
more typical of R packages), Undefined global functions or variables: data - and
more

ok

> * there' a note about compressing your data better

This is as I went back to save the original data in csv as suggested by
@RemkoDuursma above (https://github.com/ropensci/onboarding/issues/35#issuecomment-216434323) to facilitated version control of the data. I would rather keep the data as csv.

> * you can still write minimal tests even though this is a data pkg, e.g. load
data, and make sure data is of right dimensions, etc.

ok

**Comment by sckott** [comment]
Okay on the csv - since you intend to send to CRAN, they may complain about that, but you can address that then

**Comment by sckott** [comment]
@Dahaniel anything to report on this?

**Comment by Dahaniel** [comment]
see https://github.com/ropensci/onboarding/issues/34#issuecomment-315562774

**Comment by Dahaniel** [comment]
Done, please see https://github.com/ropensci/onboarding/issues/34#issuecomment-324934191

**Comment by sckott** [comment]
thanks @Dahaniel

**Comment by sckott** [comment]
Sorry for the very long wait @Dahaniel !  

Approved!  Thanks again for your submission  

* Please transfer the package  to `ropensci`- I've invited you to a team within our ropensci organization. After you accept you should be able to move it. 
* Make sure to 
    * add rOpenSci footer to README
`[![ropensci_footer](https://ropensci.org/public_images/ropensci_footer.png)](https://ropensci.org)`
    * Change any needed links, such those for CI badges
    * Travis CI should update to the new location automatically - you may have to update other CI systems manually
* Add dotgithub files, see https://github.com/ropensci/dotgithubfiles to make sure users report session info, etc. and to give guidance on how to contribute - issue and PR template probably can be used as is, but feel free to modify CONTRIBUTING as you like
* We're starting to roll out software metadata files to all ropensci pkgs via the Codemeta initiative, see  https://github.com/ropensci/codemetar/#codemetar  for how to include it in your pkg, after installing the pkg - should be easy as running `codemetar::write_codemeta()` in the root of your pkg
* Add the "Peer-reviewed" status badge to your README:

```
[![](https://badges.ropensci.org/35_status.svg)](https://github.com/ropensci/onboarding/issues/35)
```
